clang/include/clang/Basic/Attr.td

```diff
@@ -2868,6 +2868,11 @@ def M68kRTD: DeclOrTypeAttr {
   let Documentation = [M68kRTDDocs];
 }
 
+def PreserveNone : DeclOrTypeAttr {
+  let Spellings = [Clang<"preserve_none">];
```

Comment:
This should have a subjectlist set, since this is on functions, it should probably have a function subject.  Since it is target specific, it should ALSO be a target specific attribute.

Reply:
I tried to add
```
  let Subjects = SubjectList<[Function]>;
```
But it doesn't work for the code
```
  void __attribute__((preserve_none)) boo(void *ptr) {
  }

  void (__attribute__((preserve_none)) *pboo1)(void *) = boo;
```
I got the error message
```
'preserve_none' attribute only applies to functions
```

Reply:
The second example there is a function pointer, not a function you're trying to apply it to.  If you want it to work on Function Pointers as well, you need to use `FunctionLike`.

Reply:
FunctionLike works for me. Thanks!